src/core/services/compiler/compiler.spec.js

```diff
@@ -159,27 +160,135 @@ describe('$mdCompiler service', function() {
         expect(scope.myControllerAs).toBe(data.element.controller());
       }));
 
-      it('should work with bindToController', inject(function($rootScope) {
-        var called = false;
-        var data = compile({
-          template: 'hello',
-          controller: function($scope) {
+    });
+
+  });
+
+  [
+      {respectPreAssignBindingsEnabled: true},
+      {respectPreAssignBindingsEnabled: false},
+      {respectPreAssignBindingsEnabled: '"default"', equivalentTo: false}
+  ].forEach(function(options) {
+    var realRespectPreAssignBindingsEnabled = options.respectPreAssignBindingsEnabled;
+    var respectPreAssignBindingsEnabled = angular.isDefined(options.equivalentTo) ?
+      options.equivalentTo :
+      realRespectPreAssignBindingsEnabled;
+
+    describe('with respectPreAssignBindingsEnabled set to ' + realRespectPreAssignBindingsEnabled, function() {
+      var preAssignBindingsEnabledInAngularJS = angular.version.minor < 6;
+
+      beforeEach(function() {
+        module(function($mdCompilerProvider) {
+          console.log(angular.version, preAssignBindingsEnabledInAngularJS);
+          // Don't set the value so that the default state can be tested.
+          if (realRespectPreAssignBindingsEnabled === true || realRespectPreAssignBindingsEnabled === false) {
```

Comment:
I wonder why I didn't think about that having used exactly this method in source. :)